nistats/second_level_model.py

```diff
@@ -0,0 +1,451 @@
+            if design_matrix is None:
+                design = _create_second_level_design(design_df, confounds)
+
+        else:
+            design = design_matrix
+            effects_maps = first_level_input
+
+        # set design matrix, given or computed
+        if design_matrix is None:
+            self.design_matrix_ = design
+        else:
+            self.design_matrix_ = design = design_matrix
+        # check design matrix X and effect maps Y agree on number of rows
+        if len(effects_maps) != design.shape[0]:
+            raise ValueError('design_matrix X number of rows do not agree with'
+                             ' number of maps Y considered')
+        # check niimgs
+        for niimg in effects_maps:
+            check_niimg(niimg, ensure_ndim=3)
+
+        # Report progress
+        t0 = time.time()
+        if self.verbose > 0:
+            sys.stderr.write("Computing second level model. "
+                             "Go take a coffee\r")
+
+        # Learn the mask. Assume the first level imgs have been masked.
+        if not isinstance(self.mask, NiftiMasker):
+            self.masker_ = NiftiMasker(
+                mask_img=self.mask, smoothing_fwhm=self.smoothing_fwhm,
+                memory=self.memory, verbose=max(0, self.verbose - 1),
+                memory_level=self.memory_level)
+        else:
+            self.masker_ = clone(self.mask)
+            for param_name in ['smoothing_fwhm', 'memory', 'memory_level']:
+                our_param = getattr(self, param_name)
+                if our_param is None:
+                    continue
+                if getattr(self.masker_, param_name) is not None:
+                    warn('Parameter %s of the masker overriden' % param_name)
+                setattr(self.masker_, param_name, our_param)
+        self.masker_.fit(effects_maps[0])
+
+        # Fit the model
+        Y = self.masker_.transform(effects_maps)
+        if self.memory is not None:
+            mem_glm = self.memory.cache(run_glm)
+        else:
+            mem_glm = run_glm
+        labels, results = mem_glm(Y, design.as_matrix(), n_jobs=self.n_jobs)
+        # We save memory if inspecting model details is not necessary
+        if self.minimize_memory:
+            for key in results:
+                results[key] = SimpleRegressionResults(results[key])
+        self.labels_ = labels
+        self.results_ = results
+        del Y
+
+        # Report progress
+        if self.verbose > 0:
+            sys.stderr.write("\nComputation of second level model done in "
+                             "%i seconds\n" % (time.time() - t0))
+
+        return self
+
+    def compute_contrast(self, contrast_def, stat_type=None,
+                         output_type='z_score'):
+        """Generate different outputs corresponding to
+        the contrasts provided e.g. z_map, t_map, effects and variance.
+        In multi-session case, outputs the fixed effects map.
```

Comment:
this does not make sense here: no multi-session here (?)
Please update the whole docstring accordingly.